File: stalker-shop/src/main/java/com/itis/stalkershop/services/interfaces/ValidationService.java
package com.itis.stalkershop.services.interfaces;

import com.itis.stalkershop.models.UserRegister;
import com.itis.stalkershop.utils.exceptions.ErrorEntity;

import java.util.Optional;

public interface ValidationService {
    Optional<ErrorEntity> validateRegistration(UserRegister newUser);
}

File: stalker-shop/src/main/java/com/itis/stalkershop/services/interfaces/ValidationService.kt
package com.itis.stalkershop.services.interfaces;

import com.itis.stalkershop.models.UserRegister;
import com.itis.stalkershop.utils.exceptions.ErrorEntity;

import java.util.Optional;

interface ValidationService {
    // Why it returns an optional error??
    // Wouldn't it be better if it would just throw an exception instead?
    fun validateRegistration(newUser: UserRegister): Optional<ErrorEntity>
}
